### DISCUSSION TITLE
How to let PMM not create empty assets folders? Or how to remove empty assets folders via PMM?

### DISCUSSION THREAD

**Discussion by gitttt54** [meisnate12/Plex-Meta-Manager#1434] [Q&A]
Hello,

when going through the assets folder via windows explorer it takes ages to load that folder, because PMM creates very much collection-folders (which are mostly empty).
 Loading that folder takes an crazy amount of time because of the count of subfolders for each collection.

Is there a way to tell PMM to remove those empty folders after each run?
or not to create empty folders?

I don't want to run a seperate cron job with something like ` find . -type d -empty -exec rmdir "{}" \;` because I'm afraid, it'll harm PMM's work, when those two run at the same time.

Thank you!

**Comment by chazlarson** [comment]
There isn't a way to have PMM clear out the empties.

There isn't a way to have it not create empty folders since any folder it creates will be empty, waiting for you to put an asset image into it.

I have a script that can build out that folder setup, sorted into A-Z folders, if that's useful instead of having PMM create them.

To allay the fear, you could run PMM with a script that executes the find after PMM has completed.

**Reply by gitttt54** [reply]
>since any folder it creates will be empty, waiting for you to put an asset image into it.

Ok, that makes sense and seems nescessaray to me now. Thank you.

>To allay the fear, you could run PMM with a script that executes the find after PMM has completed.

I'm on Unraid and he PMM docker container starts on reboot and has PMM_TIME set to `00:00,01:00,02:00,03:00,04:00,05:00,06:00,07:00,08:00,09:00,10:00,11:00,12:00,13:00,14:00,15:00,16:00,17:00,18:00,19:00,20:00,21:00,22:00,23:00`. 
I  have 2 main configs scheduled.

If I remember the wiki and some old conversation of us correctly, I need to remove the PMM_TIME and set up cron (with the same times) to run a script starting PMM and the find afterwards. 
But then PMM will run at container start with the default config.yml too because there is no way to have the docker container start, but not PMM. (PMM will run every day at 05:00 per default but i don't want that when I have cron set up).

I need to re-read the wiki and think about that scenario again, thank you for your input!

**Reply by chazlarson** [reply]
One way to do it would be to remove the schedule you have there and replace it with a cron job that runs a script on the hour to do a manual run of PMM followed by the find operation.

Realistically I think you can just cron the `find` maybe hourly at X:58 since PMM isn't going to be doing anything with empty folders.  However, it's just going to recreate them again.

Maybe you'd want to run a separate script to build out that folder hierarchy once a day or something and turn off PMM's creation of asset folders.  If you used my script you could build it out sorted into A-Z folders and/or by library, which would reduce the load on Windows File Explorer.

> there is no way to have the docker container start, but not PMM (PMM will run every day at 05:00 per default 

This is not true.  If you start the docker container with the `--run` flag, PMM will run immediately and then the container will exit.  Without that flag [or one of its equivalents] the container will start and PMM will wait until 5AM.  There isn't a single-command mechanism to run it immediately and *also* wait until 5AM.

```
docker run --rm -v "/some/path/:/config:rw" meisnate12/plex-meta-manager --run
```
Spins up a container, runs PMM immediately, tears down and deletes the container.

```
docker run --rm -v "/some/path/:/config:rw" meisnate12/plex-meta-manager
```
Spins up a container, runs PMM which waits until 5AM to wake up and do its thing, container hangs around until you stop it, then it gets deleted.

```
docker run --rm -d -v "/some/path/:/config:rw" meisnate12/plex-meta-manager
```
Same as above, but in the background.
